mysqld_exporter.go

```diff
@@ -918,7 +918,18 @@ func scrapeGlobalVariables(db *sql.DB, ch chan<- prometheus.Metric) error {
 }
 
 func scrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
-	slaveStatusRows, err := db.Query(slaveStatusQuery)
+	var (
+		slaveStatusRows *sql.Rows
+		err error
+	)
+	// Leverage lock-free SHOW SLAVE STATUS by guessing the right suffix
+	querySuffixes := []string{" NONBLOCKING", " NOLOCK", " "}
```

Comment:
The last item can be empty string (`""`) now that this doesn't need to remember.